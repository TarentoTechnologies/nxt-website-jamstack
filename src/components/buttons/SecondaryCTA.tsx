import { Link } from "gatsby";
import React from "react";

import { labelStyles, large, secondaryBtn } from "./SecondaryCTA.module.css";

interface SecondaryCTAProps {
  label: any;
  size?: string;
  clickHandler?: (event: any) => void;
  customClasses?: any;
  btnLink?: string;
}

export const SecondaryCTA = ({
  label,
  size,
  clickHandler,
  customClasses,
  btnLink,
}: SecondaryCTAProps) => {
  return (
    <Link to={btnLink != null ? btnLink : ""} data-testid="secondary-cta">
      <div
        className={`text-center d-flex align-items-center justify-content-center 
        ${size === "large" ? large : ""} ${customClasses} ${secondaryBtn}`}
        onClick={clickHandler}
        data-aos="fade-up"
      >
        <label className={`${labelStyles}`}>{label}</label>
      </div>
    </Link>
  );
};
