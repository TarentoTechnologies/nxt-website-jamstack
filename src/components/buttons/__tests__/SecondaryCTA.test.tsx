import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import React from "react";

import { SecondaryCTA } from "../SecondaryCTA";

describe("Test if the SecondaryCTA component is working as expected", () => {
  it("<SecondaryCTA /> matches snapshot", () => {
    const component = render(
      <SecondaryCTA data-testid="secondary-cta" label="Register Now" />
    );
    expect(component.container).toMatchSnapshot();
  });

  it("<SecondaryCTA /> renders without exploding", () => {
    const component = render(
      <SecondaryCTA data-testid="secondary-cta" label="Register Now" />
    );
    expect(component.getByTestId("secondary-cta")).toBeInTheDocument();
  });
});
