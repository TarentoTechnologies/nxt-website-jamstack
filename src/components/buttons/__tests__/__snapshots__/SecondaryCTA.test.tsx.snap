// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test if the SecondaryCTA component is working as expected <SecondaryCTA /> matches snapshot 1`] = `
<div>
  <a
    data-testid="secondary-cta"
    href=""
  >
    <div
      class="text-center d-flex align-items-center justify-content-center 
         undefined secondaryBtn"
      data-aos="fade-up"
    >
      <label
        class="labelStyles"
      >
        Register Now
      </label>
    </div>
  </a>
</div>
`;
